modules/undertoneBidAdapter.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Adapter to send bids to Undertone
+ */
+
+import * as utils from 'src/utils';
+import { registerBidder } from 'src/adapters/bidderFactory';
+
+const BIDDER_CODE = 'undertone';
+const URL = '//hb.undertone.com/hb'; // //ads.undertone.com/hb
+
+export const spec = {
+  code: BIDDER_CODE,
+  isBidRequestValid: function(bid) {
+    if (bid && bid.params && bid.params.publisherId && bid.params.placementId) {
+      bid.params.publisherId = parseInt(bid.params.publisherId);
+      return true;
+    }
+  },
+  buildRequests: function(validBidRequests) {
+    const payload = {
+      'x-ut-hb-params': []
+    };
+    const timeout = window.PREBID_TIMEOUT || null;
+    const host = utils.getTopWindowLocation().host;
+    const domain = /[-\w]+\.(?:[-\w]+\.xn--[-\w]+|[-\w]{3,}|[-\w]+\.[-\w]{2})$/i.exec(host);
+
+    const pubid = validBidRequests[0].params.publisherId;
+    const REQ_URL = `${URL}?pid=${pubid}&domain=${domain}`;
+
+    validBidRequests.map(bidReq => {
+      const bid = {
+        bidRequestId: bidReq.bidId,
+        hbadaptor: 'prebid',
+        domain: domain,
+        placementId: bidReq.params.placementId,
+        publisherId: bidReq.params.publisherId,
+        sizes: bidReq.sizes,
+        timeout: timeout,
+        params: bidReq.params
+      };
+      payload['x-ut-hb-params'].push(bid);
+    });
+    return {
+      method: 'POST',
+      url: REQ_URL,
+      withCredentials: true,
+      data: JSON.stringify(payload)
+    };
+  },
+  interpretResponse: function(serverResponse, request) {
+    const bids = [];
+
+    if (serverResponse && Array.isArray(serverResponse) && serverResponse.length > 0) {
+      serverResponse.forEach((bidRes) => {
+        if (bidRes.ad && bidRes.cpm > 0) {
+          const bid = {
+            requestId: bidRes.bidRequestId,
+            bidderCode: BIDDER_CODE,
```

Comment:
`bidderCode` will be set by `bidderFactory` automatically now, this line can be dropped